presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -368,66 +371,89 @@ protected Node visitShowPartitions(ShowPartitions showPartitions, Void context)
             TableHandle tableHandle = metadata.getTableHandle(session, table)
                     .orElseThrow(() -> new SemanticException(MISSING_TABLE, showPartitions, "Table '%s' does not exist", table));
 
-            List<TableLayoutResult> layouts = metadata.getLayouts(session, tableHandle, Constraint.alwaysTrue(), Optional.empty());
-            if (layouts.size() != 1) {
-                throw new SemanticException(NOT_SUPPORTED, showPartitions, "Table does not have exactly one layout: %s", table);
+            TableLayout layout = getLayout(showPartitions, showPartitions.getTable(), showPartitions.getWhere())
```

Comment:
check that `showPartitions.getWhere()` is using only partitioned columns

Reply:
added tests for that